packages/experimental-utils/src/ts-eslint/Rule.ts

```diff
@@ -165,6 +165,13 @@ type ReportDescriptor<TMessageIds extends string> =
   ReportDescriptorWithSuggestion<TMessageIds> &
     (ReportDescriptorNodeOptionalLoc | ReportDescriptorLocOnly);
 
+/**
+ * Plugins can add their settings using declaration
+ * merging against this interface.
+ */
+// eslint-disable-next-line @typescript-eslint/no-empty-interface
+interface SharedConfigurationSettings {}
```

Comment:
let's keep the index signature so that you can access whatever properties without casting still.
Otherwise this is a big breaking change.

```suggestion
interface SharedConfigurationSettings {
  [name: string]: unknown;
}
```

Reply:
But then that lets you access arbitrary properties without complaint from typescript (at least in this area) - my thinking was that while there are pros and cons to both sides of the coin, you can *add* the index signature to the interface using declaration merging, whereas it's not possible to *remove* the index if it's included here, so that would be the better to go with since it encourages more accurate types (by requiring you to DM either an index or specific properties).

I was also thinking the break would be acceptable given previous breaks in the types, and that anyone using this specific property/type would already have to be casting.

But happy to add the index back if that's still your preference :)

Reply:
I don't think it would matter that you can access anything because any unknown things will be typed as `unknown` - so you'll be blocked from doing anything with them!

There's also the [`noPropertyAccessFromIndexSignature`](https://www.typescriptlang.org/tsconfig#noPropertyAccessFromIndexSignature) tsconfig option if you want TS to help remind you about unknown props!

This will be a breaking change other wise as `context.settings.foo` will now be a TS error unless you specifically update your code to extend the interface